airflow/dag_processing/processor.py

```diff
@@ -187,6 +188,19 @@ def _handle_dag_file_processing():
 
     def start(self) -> None:
         """Launch the process and start processing the DAG."""
+        import importlib
+
+        modules = get_airflow_modules_in(self.file_path)
+        for module in modules:
+            try:
+                importlib.import_module(module)
+            except Exception as e:
+                # only log as warning because an error here is not preventing anything from working,
+                # and if it's serious, it's going to be surfaced to the user when the dag is actually parsed.
+                self.log.warning(
+                    f"error when try to pre-import module '{module}' found in {self.file_path}: {e}"
+                )
```

Comment:
Do not use f-string in logging.

Reply:
is it for performance reasons when the log level is set higher ? Or for an other reason ?

Reply:
https://blog.pilosus.org/posts/2020/01/24/python-f-strings-in-logging/

Reply:
ok, so "performance reasons when the log level is set higher" + sentry error grouping.